### PR TITLE
Debugging`generate_gwl_tables.py` file for timestamp jumping in WL generation

### PR DESCRIPTION
**Pull request by claalmve**
Debugging timestamp jumping in warming level generation in `generate_gwl_tables.py` and also a new data product to be used in `agnostic` tools (creating csv that indexes from times to warming levels).

### CODE REVIEW

File: climakitae/data/gwl_1981-2010ref_debugged.csv
[the remark this answers is not in the document]
Comment (elehmer) [reply climakitae/data/gwl_1981-2010ref_debugged.csv:1]: Agreed, not seeing a ref

File: climakitae/util/generate_gwl_tables.py [hunk 1]
```diff
@@ -246,34 +274,87 @@ def get_gwl_table(
...
+
+    for i, model in enumerate(models):
+        print(f"\n...Model {i} {model}...\n")
+        gw_tbl, wl_data_tbl_sim = get_gwl_table(variable, model, scenarios)
+        all_gw_tbls.append(gw_tbl)
+        all_gw_data_tbls.append(wl_data_tbl_sim)
+        try:
+            all_wl_data_tbls = pd.concat([all_wl_data_tbls, wl_data_tbl_sim], axis=1)
+        except Exception as e:
+            all_wl_data_tbls.to_csv(f"../data/gwl_1850-1900ref_timeidx_{model}.csv")
```
Comment (elehmer) [inline climakitae/util/generate_gwl_tables.py:294]: You can use relative path here `data/gwl...`

File: climakitae/util/generate_gwl_tables.py [hunk 2]
```diff
@@ -246,34 +274,87 @@ def get_gwl_table(
...
+
+    ### Writing out all warming level data table for all models
+
+    # Resetting index for CESM2 since it records the dates on the 16th rather than the 15th.
+    wl_data_tbl_cesm2.index = all_gw_data_tbls[0].index
+    all_gw_data_tbls[1].index = all_gw_data_tbls[0].index
+
+    # Combining and writing out
+    all_wl_data_tbls = pd.concat(all_gw_data_tbls + [wl_data_tbl_cesm2], axis=1)
+    all_wl_data_tbls.to_csv("../data/gwl_1850-1900ref_timeidx.csv")
```
Comment (elehmer) [inline climakitae/util/generate_gwl_tables.py:306]: relative path

File: climakitae/util/generate_gwl_tables.py [hunk 3]
```diff
@@ -246,34 +274,87 @@ def get_gwl_table(
...
 
+    #####
+    # # Resetting index for CESM2 since it records the dates on the 16th rather than the 15th.
+    # wl_data_tbl_cesm2.index = all_gw_data_tbls[0].index
+    # # all_gw_data_tbls[1].index = all_gw_data_tbls[0].index
+
+    # # Combining and writing out
+    # # all_wl_data_tbls = pd.concat(all_gw_data_tbls + [wl_data_tbl_cesm2], axis=1)
+    # all_wl_data_tbls = pd.concat([all_wl_data_tbls, wl_data_tbl_cesm2], axis=1)
+    # all_wl_data_tbls.to_csv("../data/gwl_1850-1900ref_timeidx.csv")
```
Comment (elehmer) [inline climakitae/util/generate_gwl_tables.py:324]: relative path

**Review by elehmer — CHANGES_REQUESTED** [review]

File: climakitae/explore/agnostic.py [hunk 4]
```diff
@@ -38,9 +38,7 @@ def create_lookup_tables():
 def _create_time_lut(gcms):
     """Prepare lookup table for converting warming levels to times."""
     # Read in simulation vs warming levels (1.5, 2, 3, 4) table
-    df = pd.read_csv(
-        "~/src/climakitae/climakitae/data/gwl_1850-1900ref_agnostic_tools.csv"
-    )
+    df = pd.read_csv("~/src/climakitae/climakitae/data/gwl_1850-1900ref.csv")
```
Comment (claalmve) [inline climakitae/explore/agnostic.py:41]: Ignore this `read_csv`, I will fix this tomorrow with the correct relative path reading

**Review by elehmer — APPROVED** [review]
Really hard to review with a lot of the code already merged...there are a couple of comments on the previous commit of agnostic.py that you hopefully see. Let's not do this again lol. Thx

**Pull request merged**